src/components/base/JIcon/JIcon.js

```diff
@@ -2,36 +2,47 @@
 
 import React, { PureComponent } from 'react'
 import classNames from 'classnames'
+import { keyBy } from 'lodash-es'
 
-export type JIconSize = 'small' | 'medium' | 'large' | 'xlarge'
 export type JIconColor = 'white' | 'blue' | 'gray' | 'sky' | 'red' | 'black'
 
 type Props = {
   name: string,
-  size: JIconSize,
   color: ?JIconColor,
+  useFill: boolean,
 }
 
 const files = require.context('../../../public/assets/icons/sprite-pack', true, /.*\.svg$/)
-const icons = files.keys().map(x => files(x).default).reduce((result, { id, url }) => ({
-  ...result,
-  [id]: url,
-}), {})
+const icons = keyBy(
+  files.keys().map((x) => {
+    const filesArray = files(x).default
+    const sizeArray = filesArray.viewBox.split(/(\s+)/).filter(e => e.trim().length > 0)
+    /* eslint-disable prefer-destructuring, fp/no-mutation */
+    filesArray.width = sizeArray[2]
+    filesArray.height = sizeArray[3]
+    /* eslint-enable prefer-destructuring, fp/no-mutation */
+    return filesArray
+  }),
+  'id',
+)
 
 class JIcon extends PureComponent<Props> {
   static defaultProps = {
-    size: 'medium',
     color: null,
-    useСurrentСolor: false,
+    useFill: false,
```

Comment:
How do I know from outside of the icon if it uses fill or not? Looks like this depends only on icon id too.

Maybe we could skip guessing and just set up both fill and stroke color for icon? If it does not have stroke, it would have stroke-width set to 0 inside of its svg source.

Reply:
So far, I have no idea how to check it. We previously remove the fill and stroke attributes from the icons at the assembly stage. Now there are only two such icons.